activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb

```diff
@@ -131,15 +131,11 @@ def execute(sql, name = nil)
       def exec_query(sql, name = 'SQL', binds = [], prepare: false)
         result = execute(sql, name)
         @connection.next_result while @connection.more_results?
-        ActiveRecord::Result.new(result.fields, result.to_a)
+        ActiveRecord::Result.new(result.fields, result.to_a) if result
```

Comment:
If a query does not have a result, `execute` returns `nil`:

```
irb(main):002:0> conn.execute("create table t(a int)")
   (43.3ms)  create table t(a int)
=> nil
irb(main):003:0> conn.execute("insert into t values (1)")
   (19.3ms)  insert into t values (1)
=> nil
```

`exec_insert` calls `exec_query`:

https://github.com/kamipo/rails/blob/remove_not_needed_exec_insert/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb#L87-L92

```ruby
      # Executes insert +sql+ statement in the context of this connection using
      # +binds+ as the bind substitutes. +name+ is logged along with
      # the executed +sql+ statement.
      def exec_insert(sql, name, binds, pk = nil, sequence_name = nil)
        exec_query(sql, name, binds)
      end
```

This `if result` is needed.